tests/test_core.py

```diff
@@ -7687,6 +7687,11 @@ def test_stack_overflow_check(self):
     self.emcc_args += ['-DONE_BIG_STRING']
     self.do_runf(path_from_root('tests', 'stack_overflow.cpp'), 'stack overflow', assert_returncode=NON_ZERO)
 
+    # ASSERTIONS=2 implies STACK_OVERFLOW_CHECK=2
+    self.set_setting('STACK_OVERFLOW_CHECK', 0)
```

Comment:
I think you can do `clear_setting` here instead.   That will mean its not passed at all.

Reply:
Is that better or worse in this case? If the test mode sets the option, I actually do want to override that to 0, I think?

Reply:
clear_setting would remove any setting set by the test mode too.

Reply:
Actually if you explictly set `-s STACK_OVERFLOW_CHECK=0` on the command line that shouldn't that take presence over the `-s ASSERTIONS=2`?   Or if not at least it should be an error.  We don't want to ignore an explicit command line argument like that do we?

Reply:
Oh, I didn't realize that, thanks. Done.

Reply:
Oh, good point in the last comment too. Yes, it should not be ignored, fixed.